aws/s3/lock.go

```diff
@@ -10,6 +10,28 @@ type Lock struct {
 	UUID string `json:"uuid,omitempty"`
 }
 
+type UserLock struct {
+	User       string `json:"user,omitempty"`
+	LockReason string `json:"lock_reason", omitempty"`
+}
+
+func CheckUserLock(s3c aws.S3API, bucket *string, lock_path *string) (bool, *UserLock, error) {
```

Comment:
CheckUserLock should probably just error with the lock information if there is a lock I think. This will make it much easier to understand what is being returned.